### PR TITLE
 TRUS-3827: Replace UTR page with Which Identifier page for 5mld

### PR DESCRIPTION
**Pull request by BriWak**

### CODE REVIEW

File: app/models/WhichIdentifier.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  val values: List[WhichIdentifier] = List(
+    UTRIdentifier, URNIdentifier, NoIdentifier
+  )
+
+  val options: List[(RadioOption, String)] = values.map {
+    value =>
+      (
+        RadioOption("whichIdentifier", value.toString),
+        if (value != NoIdentifier) s"whichIdentifier.${value.toString}.hint" else ""
```
Comment (jackheslop96) [inline app/models/WhichIdentifier.scala:37]: Minor but `${value.toString}` is the same as `$value` (I think)

File: conf/messages.en [hunk 2]
```diff
@@ -138,6 +138,43 @@ trustHaveAUTR.heading = Does the trust have a Unique Taxpayer Reference (UTR)?
 trustHaveAUTR.checkYourAnswersLabel = Does the trust have a Unique Taxpayer Reference?
 trustHaveAUTR.error.required = Select yes if the trust has a Unique Taxpayer Reference
 
+whichIdentifier.title = Which unique identifier does the trust have?
+whichIdentifier.heading = Which unique identifier does the trust have?
+whichIdentifier.error.required = Select if the trust has a unique identifier
+whichIdentifier.utr = Unique Taxpayer Reference (UTR)
+whichIdentifier.utr.hint = This is 10 numbers, for example 12345 67890.
+whichIdentifier.urn = Unique Reference Number (URN)
+whichIdentifier.urn.hint = This is 15 numbers and characters, for example ABTRUST12345678.
```
Comment (jackheslop96) [inline conf/messages.en:147]: Me being pedantic, but possibly a question for design: characters (to me at least) includes numbers. Could this say 'alphanumeric characters' or 'numbers and letters' ?

File: conf/messages.en [hunk 3]
```diff
@@ -138,6 +138,43 @@ trustHaveAUTR.heading = Does the trust have a Unique Taxpayer Reference (UTR)?
...
+refSentByPost.title = A unique identifier has been made for the trust
+refSentByPost.heading = A unique identifier has been made for the trust
+refSentByPost.paragraph1 = The type of unique identifier for your trust depends on if the trust needs to pay tax or not.
+refSentByPost.subheading1 = Trusts that need to pay tax
+refSentByPost.paragraph2 = If the trust needs to pay tax, your unique identifier will be a Unique Taxpayer Reference (UTR).
+refSentByPost.bullet1 = A UTR is ten numbers long, for example 1234567890.
+refSentByPost.paragraph3 = The UTR will be sent to you by post. It can take 15 working days for the unique identifier to arrive. For addresses outside the UK it can take 21 working days.
+refSentByPost.subheading2 = Trusts that do not need to pay tax
+refSentByPost.paragraph4 = If the trust does not need to pay tax, your unique identifier will be a Unique Reference Number (URN).
+refSentByPost.bullet2 = A URN has 15 numbers and characters, for example ABTRUST12345678
```
Comment (jackheslop96) [inline conf/messages.en:159]: Same query as above

File: conf/messages.en [hunk 4]
```diff
@@ -138,6 +138,43 @@ trustHaveAUTR.heading = Does the trust have a Unique Taxpayer Reference (UTR)?
...
+refSentByPost.paragraph5.link = register and maintain a trust.
+refSentByPost.paragraph6 = If you do not have the URN or the UTR has not arrived by post,
+refSentByPost.paragraph6.link = contact the trusts helpline (opens in a new window or tab).
+
+refSentByPost.agent.title = The trust’s unique identifier has been sent by post
+refSentByPost.agent.heading = The trust’s unique identifier has been sent by post
+refSentByPost.agent.paragraph1 = The unique identifier for the trust has been sent to the lead trustee.
+refSentByPost.agent.paragraph2 = They will have been sent one of the following unique identifiers:
+refSentByPost.agent.bullet1 = a Unique Taxpayer Reference (UTR) which is ten numbers long, for example 1234567890
+refSentByPost.agent.bullet2 = a Unique Reference Number (URN) which has 15 numbers and characters, for example ABTRUST12345678
```
Comment (jackheslop96) [inline conf/messages.en:171]: Same query as above

File: app/views/components/input_radio_with_hint.scala.html [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            <div class="form-hint" id="hint-@{field.id}">@hint</div>
+        }
+        @field.errors.map { error =>
+            <span class="error-message" id="error-message-@field.id-input">
+                <span class="visually-hidden">@messages("site.error") </span>@messages(error.message, error.args: _*)
+            </span>
+        }
+        @for(input <- inputs) {
+            <div class="multiple-choice">
+                <input id="@{input._1.id}" type="radio" name="@{field.id}" value="@{input._1.value}" aria-describedby="@{input._1.id}.hint" @if(field.value == Some(input._1.value)){checked="checked"} @if(trackGa){data-journey-click="maintain-a-trust-frontend:click:@{input._1.id}"} />
```
Comment (jackheslop96) [inline app/views/components/input_radio_with_hint.scala.html:56]: Does `maintain-a-trust-frontend` need to be changed to `trusts-frontend` for GA tracking?

Comment (BriWak) [reply app/views/components/input_radio_with_hint.scala.html]: fixed now

File: app/models/WhichIdentifier.scala (hunk 1 above)
Comment (BriWak) [reply app/models/WhichIdentifier.scala]: fixed

**Pull request merged**